app/models/application_record.rb

```diff
@@ -16,6 +16,15 @@ def self.needs_migration?
     ActiveRecord::Migration.check_pending!
   end
 
+  # Determine whether the given search term is possibly a Primary Key (it's numeric and less than 4 bytes)
+  def self.possibly_pk?(search_term) # could add optional arg for 4 byte vs 8 byte, if needed later
+    numeric = /[\d-]+/.match(search_term).try(:[], 0) == search_term
```

Comment:
I could be missing something but this is trying to see if the whole term is numeric right? I think it would be clearer to write this as:
```suggestion
    numeric = search_term =~ /\A\d+\z/
```

or maybe more concisely:
```ruby
  MAX_PK = 2_147_483_648 
  def self.possibly_pk?(search_term)
   search_term =~ /\A\d+\z/ && search_term.to_i < MAX_PK
  end
```

Reply:
Thanks @ttoomey good point, I lifted this uncritically from client_search here: https://github.com/greenriver/hmis-warehouse/blob/47cfee51730ef5fa07ba5c0557440d4f1d1579bc/app/models/concerns/client_search.rb#L19

Do you think that both should be cleaned up? as it stands this PR doesn't touch client_search, but that was an oversight on my part; if we do decide to put this code somewhere in common per the discussion in the thread below, then I think that client_search should be updated too.

Reply:
Ah, that makes sense! The ClientSearch concern is quite complex; the current pattern is including negative numbers for some reason.

You could try DRYing it up if you have high confidence that it's not changing behavior. Personally I think leaving it unchanged also reasonable given that it is complex and a bit risky to change

Reply:
Thanks, Theron. I wouldn't say I have high confidence, so I do think I'd like to leave it unchanged if other reviewers are okay with that solution.